src/components/FileInput.js

```diff
@@ -0,0 +1,71 @@
+import { useState, useRef, useEffect } from "react";
+import PlusIcon from "../assets/ic_plus.svg";
+import XIcon from "../assets/ic_X.svg";
+
+function FileInput({ name, value, onChange }) {
+  const [previewImg, setPreviewImg] = useState(null);
+  const [showPreview, setShowPreview] = useState(false);
+
+  const inputRef = useRef(null);
+
+  const handleButtonClick = () => {
+    inputRef.current.click();
+  };
+
+  const handleFileChange = (e) => {
+    const nextValue = e.target.files[0];
+    onChange(name, nextValue);
+  };
+
+  const handleClearClick = () => {
+    const inputNode = inputRef.current;
+    if (!inputNode) return;
+
+    inputNode.value = "";
+    onChange(name, null);
+  };
+
+  useEffect(() => {
+    if (!value) return;
+
+    const nextPreviewImg = URL.createObjectURL(value);
+    setPreviewImg(nextPreviewImg);
+    setShowPreview(true);
+
+    return () => {
+      setPreviewImg();
```

Comment:
p3;
이렇게 상태를 빈 값으로 초기화하실 때는 명시적으로 null을 적어주시는게 좋습니다.
setState에 빈값을 전달하면 해당 state가 undefined가 됩니다!
```suggestion
      setPreviewImg(null);
```